src/components/Header/Header.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import StyledHeader from './styles';
+
+const Header = (props) => {
+  const {
+    title,
+    subtitle,
+    thumbnail,
+    icon,
+    alignText,
+    divider,
+    dividerColor,
+    ...attributes
+  } = props;
+  const thumbnailPath = typeof thumbnail === 'string' ? thumbnail : '';
+  const dividerClass = divider ? 'withDivider' : '';
+
+  return (
+    <StyledHeader {...attributes}>
+      {
+        thumbnail && (
+          thumbnailPath ? (
+            <StyledHeader.Thumbnail
+              thumbnailPath={thumbnailPath}
+              {...attributes}
+            />
+          ) : (
+            <StyledHeader.Thumbnail {...attributes}>
+              {thumbnail}
+            </StyledHeader.Thumbnail>
+          )
+        )
+      }
+
+      <StyledHeader.Content align={alignText}>
+        <StyledHeader.Title
+          className={dividerClass}
+          dividerColor={dividerColor}
+          {...attributes}
+        >
+          {title}
+        </StyledHeader.Title>
+
+        { subtitle &&
+          (
+            <StyledHeader.Subtitle {...attributes}>
+              {subtitle}
+            </StyledHeader.Subtitle>
+          )
+        }
+
+      </StyledHeader.Content>
+
+      { icon &&
+        (
+          <StyledHeader.Icon>
+            {icon}
+          </StyledHeader.Icon>
+        )
+      }
+
+    </StyledHeader>
+  );
```

Comment:
Talking with @Felipesdeveloper we discuss about not put conditions on components. Instead use props and [composition](https://reactjs.org/docs/composition-vs-inheritance.html) that possibilities more flexibility. Remember that components may accept arbitrary props, including primitive values, React elements, or functions.

An exemple for this implementation can be some thing like:
```
<StyledHeader {...attributes}>
  {props.thumbnail}
  {props.content}
  {props.icon}
</StyledHeader>
```
And the implementation will work like this.
```
<Header
  thumbnail={
    <Thumbnail />
  }
  content= {
   "I'm a simplem text"
  }
/>
```
Take a close look! I don't pass icon props, and this will work.
You can play with some example and see how this works on this example https://codepen.io/gaearon/pen/gwZOJp?editors=0010

Reply:
Awesome! I totally agree with this concept! I believe that like this the component turns into something more powerful and flexibe.

I like this example by material-ui card. They use this concept for many of their components.
https://material-ui.com/demos/cards/#complex-interaction

Reply:
@williamespindola could you check again? 😄